src/plugins/permissionsViewer/components/RolesAndUsersPermissions.tsx

```diff
@@ -200,24 +208,53 @@ function RoleContextMenu({ guild, roleId, onClose }: { guild: Guild; roleId: str
             aria-label="Role Options"
         >
             <Menu.MenuItem
-                id="vc-pw-view-as-role"
-                label="View As Role"
+                id="vc-copy-role-id"
+                label={i18n.Messages.COPY_ID_ROLE}
                 action={() => {
-                    const role = GuildStore.getRole(guild.id, roleId);
-                    if (!role) return;
-
-                    onClose();
-
-                    FluxDispatcher.dispatch({
-                        type: "IMPERSONATE_UPDATE",
-                        guildId: guild.id,
-                        data: {
-                            type: "ROLES",
-                            roles: {
-                                [roleId]: role
+                    Clipboard.copy(roleId);
+                }}
+            />
+
+            {(settings.store as any).unsafeViewAsRole && (
+                <Menu.MenuItem
+                    id="vc-pw-view-as-role"
+                    label="View As Role"
```

Comment:
why not go i18n all the way
```suggestion
                    label={i18n.Messages.VIEW_AS_ROLE}
```